File: src/main/java/com/github/kory33/signvote/constants/PermissionNodes.java
package com.github.kory33.signvote.constants;

/**
 * Collection of permission nodes used by SignVote plugin
 * @author Kory
 */
public class PermissionNodes {
    private static final String BASE_NODE = "signvote";

    public static final String VOTE           = BASE_NODE + ".vote";
    public static final String VOTE_MORE      = BASE_NODE + ".votemore";

    public static final String UNVOTE         = BASE_NODE + ".unvote";
    public static final String CREATE_SIGN    = BASE_NODE + ".createsign";
    public static final String CREATE_SESSION = BASE_NODE + ".createsession";
    public static final String MODIFY_SESSION = BASE_NODE + ".modifysession";

    public static final String OPEN_SESSION   = BASE_NODE + ".opensession";
    public static final String CLOSE_SESSION  = BASE_NODE + ".closesession";

    public static final String DELETE_VOTEPOINT = BASE_NODE + ".deletevotepoint";

    public static final String DELETE_SESSION = BASE_NODE + ".deletesession";

    public static final String RELOAD         = BASE_NODE + ".reload";

    public static final String SAVE           = BASE_NODE + ".save";

    public static final String LIST_SESSION   = BASE_NODE + ".listsession";

    public static final String VIEW_STATS     = BASE_NODE + ".viewstats";
}

File: src/main/kotlin/com/github/kory33/signvote/constants/PermissionNodes.kt
package com.github.kory33.signvote.constants

/**
 * Collection of permission nodes used by SignVote plugin
 * @author Kory
 */
object PermissionNodes {
    private val BASE_NODE = "signvote"

    val VOTE = BASE_NODE + ".vote"
    val VOTE_MORE = BASE_NODE + ".votemore"

    val UNVOTE = BASE_NODE + ".unvote"
    val CREATE_SIGN = BASE_NODE + ".createsign"
    val CREATE_SESSION = BASE_NODE + ".createsession"
    val MODIFY_SESSION = BASE_NODE + ".modifysession"

    val OPEN_SESSION = BASE_NODE + ".opensession"
    val CLOSE_SESSION = BASE_NODE + ".closesession"

    val DELETE_VOTEPOINT = BASE_NODE + ".deletevotepoint"

    val DELETE_SESSION = BASE_NODE + ".deletesession"

    val RELOAD = BASE_NODE + ".reload"

    val SAVE = BASE_NODE + ".save"

    val LIST_SESSION = BASE_NODE + ".listsession"

    val VIEW_STATS = BASE_NODE + ".viewstats"
}
